proxy/hdrs/HdrToken.cc

```diff
@@ -230,7 +230,10 @@ static HdrTokenFieldInfo _hdrtoken_strs_field_initializers[] = {
   {"Strict-Transport-Security", MIME_SLOTID_NONE, MIME_PRESENCE_NONE, (HTIF_MULTVALS)},
   {"Subject", MIME_SLOTID_NONE, MIME_PRESENCE_SUBJECT, HTIF_NONE},
   {"Summary", MIME_SLOTID_NONE, MIME_PRESENCE_SUMMARY, HTIF_NONE},
-  {"TE", MIME_SLOTID_TE, MIME_PRESENCE_TE, (HTIF_COMMAS | HTIF_MULTVALS | HTIF_HOPBYHOP)},
+  // Need to figure out why this cannot be handled as hop by hop.  If it is hop-by-hop
+  // the information is not propagated for gRPC
+  //{"TE", MIME_SLOTID_TE, MIME_PRESENCE_TE, (HTIF_COMMAS | HTIF_MULTVALS | HTIF_HOPBYHOP)},
+  {"TE", MIME_SLOTID_TE, MIME_PRESENCE_TE, (HTIF_COMMAS | HTIF_MULTVALS)},
```

Comment:
This partially(?) reverts #7473, and it doesn't seem like a right direction. The question should be figured out before merge.

Reply:
True.  I'd leave the comment, so we have it when debugging gRPC.  But leave the code along.  There must be another way to deal with the gRPC case. Or perhaps the other changes made for #7473 fixed the gRPC case as well.

Reply:
Thank you Susan for the history. I added `HTIF_MULTVALS` back in and added a TODO comment.